### PR TITLE
Add track names

### LINKED ISSUE #129: Show track names
This is important for, say, knowing which pileup is Normal and which is Tumor.

IGV uses a left gutter ("Sequence", "RefSeq Genes"):
screen shot 2015-05-08 at 10 22 48 am

Dalliance shows them as overlaid semi-transparent bubbles ("DNase I"):
screen shot 2015-05-08 at 10 20 48 am

@jaclynperrone I'd love your thoughts on this! :)

### PR DESCRIPTION
**Pull request by danvk**
Here's what it looks like:

screen shot 2015-06-16 at 2 39 15 pm

This reworks `pileup.css` to use flexbox, which worked very well in Cycledash. You can see the shell of the layout in [this codepen][1].

@jaclynperrone any design feedback would be much appreciated here!

Fixes #129 

[1]: http://codepen.io/anon/pen/VLzbBe?editors=110

[Review on Reviewable](https://reviewable.io/reviews/hammerlab/pileup.js/195)

### CODE REVIEW

File: style/pileup.css [hunk 1]
```diff
@@ -4,11 +4,47 @@
...
+}
+
+.pileup-root > .track {
+  display: flex;
+  flex-direction: row;
+}
+.track-label {
+  flex: 0 0 100px;  /* fixed-width track labels */
+  text-align: right;
+  font-family: Helvetica Neue, Helvetica, Arial, sans;
```
Comment (jaclynperrone) [inline style/pileup.css]: Let's use the 'freight-sans' typekit font we use in Cycledash (unless you want to keep it universal)
`font-family: "freight-sans-pro", verdana, helvetica, sans-serif;`

If you want to keep it Helvetica:
Multi-word fonts are usually wrapped in quotation marks, and 'sans-serif' is used as the catch-all:
`font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;`

File: style/pileup.css [hunk 2]
```diff
@@ -24,7 +60,7 @@
   text-anchor: middle;
 }
 .pair.loose text {
-  font-size: 28;
+  font-size: 24;
```
Comment (jaclynperrone) [inline style/pileup.css]: I think you may need a unit here? Preferably `em`

File: style/pileup.css [hunk 3]
```diff
@@ -24,7 +60,7 @@
...
 }
 .pair.tight text {
   font-size: 12;
```
Comment (jaclynperrone) [inline style/pileup.css]: Same comment as above. Do you need to specify a unit?

File: style/pileup.css (hunk 1 above)
Comment (danvk) [inline style/pileup.css]: Nice catch! I quoted correctly & changed `sans` to `sans-serif`.

I'd prefer to keep TypeKit out of core pileup.js. But users of pileup (like Cycledash) are free to use it in their own stylesheets.

File: style/pileup.css (hunk 2 above)
Comment (danvk) [inline style/pileup.css]: Usually that would be true, but in this case I'm styling an SVG element. SVG has its own notion of coordinates, which are unit-free. This corresponds to 24px.

**Comment by jaclynperrone** [comment]
Looking good! Added some line-notes. And a couple other little things:

- Can we have the gene name font be defaulted to Helvetica Neue as well? You could also get away with making it a little smaller. (And I'm debating whether or not it should be left-aligned.)
`font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;`

- I would also try moving the gene name above the blue track. Right now it looks disconnected.

- The blue boxes in the variants panel are a little confusing to me. They look like they are supposed to contain something, but are showing up empty (simply adding a fill would fix this). They are also begging to be extended down through the pileup, kind of like how we talked about here #180. It would help keep a user's place as they scrolls through the BAM. What do you think? (Also sounds like it's for a different PR)

**Comment by danvk** [comment]
I changed the gene name font to Helvetica Neue. I think the other items are for a follow-up PR.

The gene track could definitely benefit from some design thinking. I tried to mirror IGV, which centers the gene name under the portion of the transcript that's visible, but it doesn't make as much sense visually w/o borders separating all the tracks.

**Comment by jaclynperrone** [comment]
Yes! I forgot to mention the borders thing. It would be good to use borders to separate the tracks. But yes, all these things can be for another time.

**Pull request merged**